code/game/objects/items/trash.dm

```diff
@@ -8,6 +8,12 @@
 	desc = "This is rubbish."
 	resistance_flags = FLAMMABLE
 
+/obj/item/trash/decompile_act(obj/item/matter_decompiler/C, mob/user)
+	C.stored_comms["metal"] += 1
+	C.stored_comms["plastic"] += 2
+	qdel(src)
+	return TRUE
```

Comment:
Put `TRUE` on the `.` value before the qdel instead

Reply:
IE:
. = TRUE
qdel(src))